src/LaunchDarkly.EventSource/ErrorStrategy.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using LaunchDarkly.EventSource.Events;
+using LaunchDarkly.EventSource.Exceptions;
+
+namespace LaunchDarkly.EventSource
+{
+    /// <summary>
+    /// An abstraction of how to determine whether a stream failure should be thrown to the
+	/// caller as an exception, or treated as an event.
+    /// </summary>
+    /// <para>
+    /// Instances of this class should be immutable.
+    /// </para>
+	/// <seealso cref="ConfigurationBuilder.ErrorStrategy(ErrorStrategy)"/>
+    public abstract class ErrorStrategy
```

Comment:
Equivalent to the [Java version](https://github.com/launchdarkly/okhttp-eventsource/blob/main/src/main/java/com/launchdarkly/eventsource/ErrorStrategy.java).